HIP/hip-1056.md

```diff
@@ -0,0 +1,1822 @@
+    * user transaction that spawned it.
+    */
+    proto.Timestamp parent_consensus_timestamp = 3;
+
+    /**
+    * A schedule that executed this transaction, if this transaction
+    * was scheduled.
+    * <p>
+    * This value SHALL NOT be set unless this transaction result represents
+    * the result of a _scheduled_ child transaction.
+    */
+    proto.ScheduleID schedule_ref = 4;
+
+    /**
+    * The actual transaction fee charged, not the original transactionFee value
+    * from TransactionBody.
+    */
+    uint64 transaction_fee_charged = 5;
+
+    /**
+    * All hbar transfers as a result of this transaction, such as fees, or
+    * transfers performed by the transaction, or by a smart contract it calls, or
+    * by the creation of threshold records that it triggers.
+    */
+    proto.TransferList transfer_list = 6;
+
+    /**
+    * All Token transfers as a result of this transaction.
+    */
+    repeated proto.TokenTransferList token_transfer_lists = 7;
+
+    /**
+    * All token associations implicitly created while handling this transaction.
+    */
+    repeated proto.TokenAssociation automatic_token_associations = 8;
+
+    /**
+    * List of accounts with the corresponding staking rewards paid as a result of
+    * a transaction.
+    */
+    repeated proto.AccountAmount paid_staking_rewards = 9;
+
+    /**
+    * Congestion pricing multiplier at the time the transaction was executed.
+    */
+    uint64 congestion_pricing_multiplier = 10;
+}
+```
+
+#### Transaction Output
+
+Output data from a transaction data that is neither present in the transaction nor stored in state. Not all transactions
+have output, each one that does has a corresponding output message type with any data it needs to include.
+
+For example a `CryptoTransfer` transaction input will result in both a `TransactionResult` and a `CryptoTransferOutput`
+to cover the outputs of the network.
+
+
+```protobuf
+message TransactionOutput {
+    oneof transaction {
+        /**
+        * Output from a contract call transaction.
+        */
+        ContractCallOutput contract_call = 1;
+
+        /**
+        * Output from a contract create transaction.
+        */
+        ContractCreateOutput contract_create = 2;
+
+        /**
+        * Output from a crypto transfer transaction.
+        */
+        CryptoTransferOutput crypto_transfer = 3;
+
+        /**
+        * Output from an ethereum transaction.
+        */
+        EthereumOutput ethereum = 4;
+
+        /**
+        * Output from a schedule create transaction that executed
+        * immediately on creation.
+        */
+        ScheduleCreateOutput schedule_create = 5;
+
+        /**
+        * Output from a schedule sign transaction that resulted in
+        * executing the scheduled transaction.
+        */
+        ScheduleSignOutput schedule_sign = 6;
+
+        /**
+          * Output from a token airdrop transaction.
+          */
+        TokenAirdropOutput token_airdrop = 7;
```

Comment:
no `TokenClaimAirdropOutput`? Image such an output proto message should have `assessed_custom_fees` and `token_transfer_lists`.

Reply:
Airdrop is special, claim does not charge fees, because the entity that sent the airdrop is required to pre-pay custom fees.
I _believe_ the state changes covers all transfers as a result.